python/ray/util/client/ray_client_helpers.py

```diff
@@ -62,3 +63,36 @@ def _CloudPicklerReducer(obj):
 
         # construct a reducer
         pickle.CloudPickler.dispatch[cls] = _CloudPicklerReducer
+
+
+@contextmanager
+def connect_to_client_or_not(connect_to_client: bool):
+    """Utility for running test logic with and without a Ray client connection.
+
+    If client_connect is True, will connect to Ray client in context.
+    If client_connect is False, does nothing.
+
+    How to use:
+    Given a test of the following form:
+
+    def test_<name>(args):
+        <initialize a ray cluster>
+        <use the ray cluster>
+
+    Modify the test to
+
+    @pytest.mark.parametrize("connect_to_client", [False, True])
+    def test_<name>(args, connect_to_client)
+    <initialize a ray cluster>
+    with connect_to_client_or_not(connect_to_client):
+        <use the ray cluster>
+
+    Parameterize the argument connect over True, False to run the test with and
+    without a Ray client connection.
+    """
+
+    if connect_to_client:
```

Comment:
I think this entire file is used only for testing. I don't know if there's a good reason it's here instead of in the test directory.

Will think about if it could be replaced with a fixture -- would require restructuring the existing placement group tests a bit and maybe modifying some of the existing fixtures in conftest.py. 
But maybe that could help make the diff smaller.